migration_tools.drush.inc

```diff
@@ -248,6 +263,67 @@ function drush_migration_tools_mt_migrate_file_folders($organization) {
 }
 
 /**
+ * Drush command callback for migration-tools-html-file-list.
+ *
+ */
+function drush_migration_tools_html_file_list($directory) {
+  // Set up options
+  $format = drush_get_option('format');
+  $outfile = drush_get_option('outfile');
+  $mask = '/.*\.htm(l)?$/i';
+
+  switch ($format) {
+    case "csv":
+      $delimiter = ", ";
+      $newline = "\n";
+      break;
+
+    default:
+      $delimiter = "\t";
+      $newline = "\n";
+  }
+
+  // Fetch file list
+  $options = array();
+  $files = file_scan_directory($directory, $mask, $options);
+
+  // Generate output
+  $output = "Path" . $delimiter . "File Name" . $delimiter . $newline;
+  foreach ($files as $file) {
+    // Truncate the file name from the path, as we're putting that in a separate column. 
+    $fullpath = explode("/", $file->uri, -1);
+    $output .= implode("/", $fullpath) . $delimiter;
+
+    $output .= $file->filename . $delimiter;
+    $output .= $newline;
+  }
+
+  // Attempt to write results to file if requested, otherwise output to terminal.
+  if ($outfile) {
+    $handle = fopen($outfile, "w");
+
+    if(!$handle) {
+      $message = "There was an error opening {$outfile} for writing, verify the path is writable and try again.";
+    }
+    else {
+      $written = fwrite($handle, $output);
+
+      if ($written === FALSE) {
+        $message = "There was an error writing to {$outfile}, verify the path is writable and try again.";
+      }
+      else {
+        $message = "File scan of {$directory} complete, saved results as {$outfile}";
+      }
+    }
+    drush_log($message);
```

Comment:
I don;t think we want to output all three possible messages the same.   One should have a status of 'success' , while the others need a status of 'error'.